subscription.tf

```diff
@@ -9,10 +19,7 @@ module "observe_cloudwatch_logs_subscription" {
   filter_name    = var.name
   filter_pattern = var.subscribed_log_group_filter_pattern
 
-  log_group_matches = concat(
-    [aws_cloudwatch_log_group.group.name, format("/aws/lambda/%s", var.name)], # Note: Data from firehose will go to itself. There is a cycle.
```

Comment:
It's a bit awkward for users to have to reverse engineer these names and pass them in to get the logs. Maybe justifies another boolean variable, a less badly named `subscribe_to_own_logs`.

Reply:
Agreed on the awkwardness, but I wasn't terribly fond of another variable either.

In hindsight, we should have retained explicit subscriptions for these log groups, since they are managed by us. The two log group names should have been concatenated to the exclusion, rather than the inclusion list. However, we can't do this _and_ explicitly subscribe the log groups in the same change, since that is virtually guaranteed to run into a race condition where we attempt to add a subscription to an already subscribed log group 🤦 

I'll update the commit to drop support for "auto" subscription for these log groups, and defer explicit subscription for a future commit. In the interim, we won't receive this data, which is unfortunate for debugging purposes but not critical.

Reply:
> In hindsight, we should have retained explicit subscriptions for these log groups

Yeah, was gonna suggest this as well. Some of the AWS services that have automatic log groups and don't accept a custom one as input will happily use an existing log group as long as it matches the expected name. I've done this for EKS I think, maybe others. So hopefully would help solve the Lambda issue.